src/modules/commander/Commander.cpp

```diff
@@ -1739,14 +1739,21 @@ Commander::run()
 
 		/* update subsystem info which arrives from outside of commander*/
 		do {
-			if (subsys_sub.updated()) {
+			if ( (updated=subsys_sub.updated()) ) {
```

Comment:
An assignment in a condition rings a bell for me.
what about a while loop since the condition is anyways at the beginning?

Reply:
This was fixed in a separate PR https://github.com/PX4/Firmware/pull/13121

Also there is one more PR that was a result of the discussion in this PR and resolves a bug mentioned here (https://github.com/PX4/Firmware/pull/13119)

* By the way this assignment in a condition loop works properly, also tested here.